apps/frontend/src/pages/tasks/libs/components/deadline/libs/helpers/update-countdown/update-countdown.helper.ts

```diff
@@ -0,0 +1,20 @@
+import { COUNTDOWN_EXPIRED } from "../../constants/constants.js";
+import { type CountdownUpdateHandlingData } from "../../types/types.js";
+import { calculateCountdown } from "../calculate-countdown/calculate-countdown.helper.js";
+
+const updateCountdown = ({
+	deadline,
+	onExpire,
+	setCountdown,
+	setIsExpired,
+}: CountdownUpdateHandlingData): void => {
+	const deadlineCountdown = calculateCountdown(deadline);
+	setCountdown(deadlineCountdown);
+
+	if (deadlineCountdown === COUNTDOWN_EXPIRED) {
+		setIsExpired(true);
+		onExpire();
+	}
```

Comment:
I don't think we need this helper. Let's keep everything related to the state in a component. Using `calculateCountdown` helper should be enough